### ISSUE TITLE
Adding new hyperparameters should be supported

### ISSUE THREAD

**Issue by wchargin** [tensorflow/tensorboard#2014]
Adding new hyperparameters should be supported

c/o @omoindrot, [who says][1]:

> 3\. Adding new hyperparameters is not fully supported. If I add a new parameter `new` and re-create the event file for the HParams tab, the previous runs that don't contain `new` will all disappear if I choose to display `new`.
>     Ideally I would be able to add values to my old summaries, but I couldn't find a way to do that.
>     For instance if I add a `batch_size` parameter, I would need add the default value `32` to all previous runs.
>     In general it would be great to be able to manipulate old summaries easily.

One approach would be to modify the [`HParamInfo` proto][proto] to
include an optional `default_value` attribute, which would be set
whenever a new hyperparameter is created and all previous runs had a
fixed value for the parameter (which should be the common case). Users
certainly shouldn’t have to modify the old event files for data from
previous runs to still be useful.

Ideally, the visualizations should be able to show partial data for runs
where one or more hyperparameters is missing: the table view could show
an empty cell to denote a missing value, the parallel coordinates view
could omit the line segments through an axis with missing data, and the
scatter plot matrix could simply omit some points.

[1]: https://github.com/tensorflow/tensorboard/issues/1998#issuecomment-472775470
[proto]: https://github.com/tensorflow/tensorboard/blob/25dc3e8ceb78ff8867e4adbea06a590089dacc9d/tensorboard/plugins/hparams/api.proto#L70

**Comment by shashvatshahi1998** [comment]
@wchargin  I want to work on this issue can you please guide me where to start and how to proceed.

**Comment by shashvatshahi1998** [comment]
@wchargin for identifying a new hparam can we distinguish using its id which is there in message HParamInfo(String name)

**Comment by wchargin** [comment]
Hi @shashvatshahi1998—thanks for your interest. There’s still some
design work to be done here, in the context of larger issues like #1998;
I don’t think that we have enough information at this time to select a
particular implementation.

**Comment by PedroUria** [comment]
Hi @wchargin @shashvatshahi1998. Has there been any progress on this topic? It would be a really useful feature to have. If not, could you point me to how to modify the event files manually?

**Comment by MaLiN2223** [comment]
Are there any updates on the topic? It would be great if we could get some information on a workaround
